src/aiida/cmdline/commands/cmd_computer.py

```diff
@@ -595,24 +595,59 @@ def computer_test(user, print_traceback, computer):
 
 
 @verdi_computer.command('delete')
+@click.option('--force', is_flag=True, help='Force deletion of computers with running nodes without confirmation')
 @arguments.COMPUTER()
 @with_dbenv()
-def computer_delete(computer):
+def computer_delete(computer, force):
     """Delete a computer.
 
     Note that it is not possible to delete the computer if there are calculations that are using it.
     """
     from aiida import orm
     from aiida.common.exceptions import InvalidOperation
+    from aiida.orm import Computer, Node
+    from aiida.orm.querybuilder import QueryBuilder
+    from aiida.tools import delete_nodes
 
     label = computer.label
 
+    # Check in the command whether the computer has any associated nodes.
+    builder = QueryBuilder()
+    builder.append(Computer, filters={'label': label}, tag='computer')
+    builder.append(Node, with_computer='computer')
+    results = builder.all()
+
+    # Extract the PKs from the results
+    node_pks = [result[0].pk for result in results]
+    tot_pks = len(node_pks)
+    existing_node = True if tot_pks > 0 else False
+
+    # Confirm deletion of existing node (if any exist)
+    if existing_node:
+        if force:
+            pass
+        elif click.confirm(
+            (
+                f'This computer has {tot_pks} associated nodes, '
+                'are you sure you want to delete this computer and its associated nodes?'
+            ),
+            abort=True,
+        ):
+            pass
+
+        _, was_deleted = delete_nodes(node_pks, dry_run=False)
```

Comment:
Can an exception be thrown here? Should we perhaps catch it just as with the computer delete statement?

Reply:
The thing is that if we don't catch the exception, it will show a scary traceback to the user. It would be better to catch it and print the error with `echo_critical` instead. But it is true that there is not a well defined exception that we can expect there. It may be any exception depending on the storage backend, so we would need a broad exception catch. This is not the nicest but can be justified in some cases. But fine, let's leave this for now